### PR TITLE
Validation: Nicer parameter list

### PR DESCRIPTION
**Pull request by lord-haffi**
I modified how the framework interprets the validator functions. Now you can name the parameters how you want but you have to specify a mapping to indicate to which field in the dataset the parameters correspond to. The new feature: You can specify the fields with any "attribute depth". I.e. you can define a parameter `x` which corresponds e.g. to a field `y` of an object `Wrapper` which is hold by the dataset as attribute `z`. Then, you just have to define the `parameter_map` as `{"x": "z.y"}`.
It's easiest if you look at the tests first, especially the two new tests at the end of the file.

### CODE REVIEW

**Comment by hf-kklein** [comment]
does this overlap with #44?

**Comment by lord-haffi** [comment]
> does this overlap with #44?

Jein. #44 wird durch diesen PR obsolet, weil ich mir eine bessere Lösung für das Problem überlegt hab.

**Review by hf-kklein — APPROVED** [review]
lgtm ;) ich glaube ich schau es mir dann in der praxis genauer an. bitte das hier noch beachten mit der setup.cfg: https://github.com/Hochfrequenz/bo4e_migration_framework/pull/46/files#r1140073338

File: requirements.in [hunk 1]
```diff
@@ -1,3 +1,4 @@
 attrs
 bo4e
 pydantic
+typeguard
```
Comment (hf-kklein) [inline requirements.in:4]: das hier musst du noch ind er setup.cfg eintragen

Comment (lord-haffi) [reply requirements.in:4]: Ah, ich vergesse das immer wieder, Mist ^^

**Pull request merged**